src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -295,13 +444,15 @@ export async function updateUBAClient(
 
       // Grab all bundles for this chain. This logic is isolated into a function that we can unit test.
       // The bundles are returned in ascending order.
-      const bundles = getMostRecentBundles(
+      const bundles = getMostRecentBundleBlockRanges(
         chainId,
         maxBundleStates,
         latestHubPoolBlockNumber,
         hubPoolClient,
         spokePoolClients
       );
+      // Make the last bundle to cover until the last spoke client searched block
+      bundles[bundles.length - 1].end = spokePoolClients[chainId].latestBlockSearched;
```

Comment:
This is the main feature add: extend latest block range to cover last client block searched

Reply:
This looks to be excellent. We have the added bonus here that our update function completely wipes the state so any update function will reset this boundary with no issues.